custom_components/bmw_connected_drive/sensor.py

```diff
@@ -142,15 +272,60 @@ def unit_of_measurement(self) -> str:
     @property
     def device_state_attributes(self):
         """Return the state attributes of the sensor."""
-        return {
+        vehicle_all_trips = self._vehicle.state.all_trips
+        vehicle_charging_profile = self._vehicle.state.charging_profile
+        vehicle_last_destinations = self._vehicle.state.last_destinations
+        result = {
             "car": self._vehicle.name,
             ATTR_ATTRIBUTION: ATTRIBUTION,
         }
+        if self._service == ALL_TRIPS:
+            attr = getattr(vehicle_all_trips, self._attribute)
+            if self._attribute in ("average_combined_consumption", "average_electric_consumption",
+                "average_recuperation", "chargecycle_range", "total_electric_distance"):
+                result['community_average'] = attr.community_average
```

Comment:
I know the HA dev team is changing from using attributes to using them as sensors in case.
So my suggestion is that you have another look at this. And you can disable sensors by default if there are too many and then a user who wants to use it can enable it again.

See the screenshots from Discord for the attributes and sensor discussion for the mobile apps.

![Screenshot_1](https://user-images.githubusercontent.com/11230573/99317520-7cc51c80-2866-11eb-9d57-b4f07243797c.png)
![Screenshot_2](https://user-images.githubusercontent.com/11230573/99317526-7e8ee000-2866-11eb-8a15-3c930a485e45.png)
![Screenshot_3](https://user-images.githubusercontent.com/11230573/99317776-ddecf000-2866-11eb-91a0-d80ff2889e54.png)

Reply:
I understand the discussion, but sometimes you want to store extra info within a sensor. This PR has plenty examples of data that you won't want as extra sensors but as extra info in the form of attributes. Take for example (in this PR) 'average_combined_consumption'; is has extra attributes concerning this value for community average, high, low, user_average and so on....